Check for jumps to preludes by default.

Rot127 opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

Remove the `tailcall` option and always check for it.
It also adds the possibility to mark instructions as `RZ_ANALYSIS_OP_TAIL`, if they are part of a tail call.

CFGs and iCFGs are not touched, they are flow wise still correct. But ignore function structures. So a `tail call` is not categorized as return for them.
This is better though, than attempting to encode abstract concepts (tail calls) into them which don't match instruction properties (jump != return, but it is conceptually). 

Others:

- Remove invalid code.
- Fix golang tests: Accept to find less strings, to not increase analysis depth.
  Because tail jumps are now correctly marked as end of a function,
  they are not always followed deeper with the default analysis depth of 1.
  Hence, less strings are found.
- Fix `pde` tests because they were wrong
  `rz_analysis_op_is_eob()` did not applied the mask onto op.type,
  so it checked against the wrong flags (especially for conditional jumps).
  Because the `eob` check was faulty the execution flow was different.

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

All green and for Hexagon, x86 added one.

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

Rot127 committed:
Check for jumps to preludes by default and follow tail calls.

Rot127 committed:
Fix test.exe tests. They were text book tail calls.

┌ root.0040103b();
│           0x0040103b      cmp   ecx, dword [0x40d004]
│       ┌─< 0x00401041      jne   0x401045
│       │   0x00401043      repz  ret
└      ┌└─> 0x00401045      jmp   root.00401335

XVilka commented on test/prj/v12-types-empty.rzdb:
```diff
@@ -206,7 +206,6 @@ analysis.jmp.indir=false
 analysis.jmp.mid=true
 analysis.jmp.ref=true
 analysis.jmp.retpoline=true
-analysis.jmp.tailcall=0
```
Don't do that - implement migration and a new project version bump instead.

XVilka reviewed (changes requested):
I like the idea but we definitely need at least a couple more tests for different architectures - ARM64 and x86_64, for example.

Rot127 committed:
Add a tail call test for x86

Rot127 committed:
Document rz_analysis_op_is_eob

Rot127 committed:
Attempt to remove seemingly irrelevant code.

Rot127 committed:
Make code more consice.

Rot127 committed:
Fix and simplify checks for op.type

Rot127 committed:
OR the TAIL property into the op.type if applicable.

Rot127 committed:
Fix test: Blocks were incorrectly assigned to function at 0x401000.

Rot127 committed:
Fix test: Function doesn't access a var at stack - 0x60

Rot127 committed:
Fix test: The match_suffix() doesn't have a saved prelude.

Rot127 committed:
Fix test: New name is the actual symbol name.

Rot127 committed:
Fix tests: Main functions get indeed called via a tail call.

Rot127 committed:
Fix test: Accept less string found to not increase analysis depth.

Because tail jumps are now correctly marked as end of a function,
they are not always followed deeper with the default analysis depth of 1.
Hence, less strings are found.

Rot127 committed:
Revert removal of options bound to version.

Rot127 committed:
Fix tests because they were wrong:

rz_analysis_op_is_eob() did not applied the mask onto op.type,
so it checked against the wrong flags (especially for conditional jumps).
Because the eob check was faulty the execution flow was different.

Rot127 committed:
Add tail call test for AArch64
